src/Promitor.Integrations.Sinks.Atlassian.Statuspage/AtlassianStatusPageClient.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Flurl;
+using GuardNet;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Promitor.Core;
+using Promitor.Integrations.Sinks.Atlassian.Statuspage.Configuration;
+
+namespace Promitor.Integrations.Sinks.Atlassian.Statuspage
+{
+    public class AtlassianStatuspageClient
+    {
+        private const string MetricRequestFormat = "{{\"data\": {{\"timestamp\": {0},\"value\": {1}}}}}";
+        private const string ApiUrl = "https://api.statuspage.io/v1";
+
+        private readonly IOptionsMonitor<AtlassianStatusPageSinkConfiguration> _sinkConfiguration;
+        private readonly ILogger<AtlassianStatuspageClient> _logger;
+        private readonly IHttpClientFactory _clientFactory;
+        private readonly IConfiguration _configuration;
+
+        public AtlassianStatuspageClient(IHttpClientFactory clientFactory, IConfiguration configuration, IOptionsMonitor<AtlassianStatusPageSinkConfiguration> sinkConfiguration, ILogger<AtlassianStatuspageClient> logger)
```

Comment:
Just a suggestion, but you could potentially pass an `IHttpClient` into this instead of the factory. If you do that, you can make use of the type-safe methods for registering a factory. So instead of using the constants, you can basically do something like:

```csharp
services.AddHttpClient<AtlassianStatuspageClient>(...);
```

What that does is configures an IHttpClient for `AtlassianStatuspageClient`, and also registers `AtlassianStatuspageClient` as transient in the DI container. The DI container then handles passing the correct client into your object.

Here's the docs: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-3.1#typed-clients

Reply:
Huh, I did not know that, thanks for the tip! Thought factory was the way to go; these things change all the time 🙈

Reply:
They both achieve the same thing, so it's really, really minor. The reason I prefer passing in the client rather than the factory is because it simplifies the code, and makes stuff like unit testing a bit simpler (since you just need to create a client, not a factory and client to pass in).

It only works if the object you're passing the client into is transient, otherwise you end up reusing the same client. I just spotted it because we were investigating this in work a few weeks ago, so lucky timing!